lib/jobs/install-os.js

```diff
@@ -148,24 +154,76 @@ function installOsJobFactory(
     };
 
     /**
+     * @memberof InstallOsJob
+     *
+     * Convert the installDisk to correct format
+     */
+    InstallOsJob.prototype._convertInstallDisk = function() {
+        var self = this;
+        var disk = self.options.installDisk;
+
+        //If disk is empty then do nothing.
+        //If disk is string, it means user directly input the drive wwid, so don't need conversion.
+        if ((!disk && disk !== 0) || _.isString(disk)) {
+            return;
+        }
+
+        if (!_.isNumber(disk)) {
+            throw new Error('The installDisk format is not correct');
```

Comment:
Just a little nitpick: I generally like to normalize what comes out of promise returning functions, so doing a `Promise.reject(new Error(...))`.